.github/scripts/download_triton_packages.py

```diff
@@ -0,0 +1,209 @@
+        arch = platform.machine()
+        if arch == "x86_64":
+            arch = "x64"
+        system_suffix = f"macos-{arch}"
+    elif system == "Linux":
+        if arch == "arm64":
+            system_suffix = "ubuntu-arm64"
+        else:
+            vglibc = tuple(map(int, platform.libc_ver()[1].split(".")))
+            vglibc = vglibc[0] * 100 + vglibc[1]
+            system_suffix = "ubuntu-x64" if vglibc > 217 else "centos-x64"
+    else:
+        return Package(
+            "llvm",
+            "LLVM-C.lib",
+            "",
+            "LLVM_INCLUDE_DIRS",
+            "LLVM_LIBRARY_DIR",
+            "LLVM_SYSPATH",
+        )
+    # use_assert_enabled_llvm = check_env_flag("TRITON_USE_ASSERT_ENABLED_LLVM", "False")
+    # release_suffix = "assert" if use_assert_enabled_llvm else "release"
+    llvm_hash_file = open("../cmake/llvm-hash.txt")
+    rev = llvm_hash_file.read(8)
+
+    name = f"llvm-{rev}-{system_suffix}"
+    url = f"https://tritonlang.blob.core.windows.net/llvm-builds/{name}.tar.gz"
+    return Package(
+        "llvm", name, url, "LLVM_INCLUDE_DIRS", "LLVM_LIBRARY_DIR", "LLVM_SYSPATH"
+    )
+
+
+def get_llvm_package_info_rocm() -> Package:
+    system = platform.system()
+    arch = platform.machine()
+    if arch == "aarch64":
+        arch = "arm64"
+    if system == "Darwin":
+        system_suffix = "apple-darwin"
+        arch = platform.machine()
+    elif system == "Linux":
+        vglibc = tuple(map(int, platform.libc_ver()[1].split(".")))
+        vglibc = vglibc[0] * 100 + vglibc[1]
+        linux_suffix = "ubuntu-18.04" if vglibc > 217 else "centos-7"
+        system_suffix = f"linux-gnu-{linux_suffix}"
+    else:
+        return Package(
+            "llvm",
+            "LLVM-C.lib",
+            "",
+            "LLVM_INCLUDE_DIRS",
+            "LLVM_LIBRARY_DIR",
+            "LLVM_SYSPATH",
+        )
+    use_assert_enabled_llvm = False
+    release_suffix = "assert" if use_assert_enabled_llvm else "release"
+    name = f"llvm+mlir-17.0.0-{arch}-{system_suffix}-{release_suffix}"
+    version = "llvm-17.0.0-c5dede880d17"
+    url = f"https://github.com/ptillet/triton-llvm-releases/releases/download/{version}/{name}.tar.xz"
+    # FIXME: remove the following once github.com/ptillet/triton-llvm-releases has arm64 llvm releases
+    if arch == "arm64" and "linux" in system_suffix:
+        url = f"https://github.com/acollins3/triton-llvm-releases/releases/download/{version}/{name}.tar.xz"
+    return Package(
+        "llvm", name, url, "LLVM_INCLUDE_DIRS", "LLVM_LIBRARY_DIR", "LLVM_SYSPATH"
+    )
+
+
+def get_llvm_package_info(build_rocm: bool) -> Package:
+    if build_rocm:
+        return get_llvm_package_info_rocm()
+    else:
+        return get_llvm_package_info_cuda()
+
+
+def open_url(url: str) -> Any:
+    user_agent = (
+        "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/119.0"
+    )
+    headers = {
+        "User-Agent": user_agent,
+    }
```

Comment:
Why do we need this? What's wrong with curl as user agent?

Reply:
Not sure, this is how it's written in triton. 
https://github.com/openai/triton/blob/51ecb6f09db45fcc5f216798177f8393b0d92689/python/setup.py#L158